include/verilated_funcs.h

```diff
@@ -549,23 +553,24 @@ static inline QData VL_EXTENDS_QQ(int, int lbits, QData lhs) VL_PURE {
 static inline WDataOutP VL_EXTENDS_WI(int obits, int lbits, WDataOutP owp, IData ld) VL_MT_SAFE {
     const EData sign = VL_SIGNONES_E(lbits, static_cast<EData>(ld));
     owp[0] = ld | (sign & ~VL_MASK_E(lbits));
-    for (int i = 1; i < VL_WORDS_I(obits); ++i) owp[i] = sign;
+    VL_MEMSET_W(owp + 1, sign, VL_WORDS_I(obits) - 1);
     return owp;
 }
 static inline WDataOutP VL_EXTENDS_WQ(int obits, int lbits, WDataOutP owp, QData ld) VL_MT_SAFE {
     VL_SET_WQ(owp, ld);
     const EData sign = VL_SIGNONES_E(lbits, owp[1]);
     owp[1] |= sign & ~VL_MASK_E(lbits);
-    for (int i = VL_WQ_WORDS_E; i < VL_WORDS_I(obits); ++i) owp[i] = sign;
+    VL_MEMSET_W(owp + VL_WQ_WORDS_E, sign, VL_WORDS_I(obits) - VL_WQ_WORDS_E);
```

Comment:
After quick confirmation by experiment, compilers don't seem optimize this, as they reasonably doesn't understand that the sign can be applied word-by-word instead of byte-by-byte. The compilers do seem to handle 0xff as a constant efficiently, which was impressive.  So...

```suggestion
    if (sign) {
        VL_MEMSET_W(owp + VL_WQ_WORDS_E, 0xff, VL_WORDS_I(obits) - VL_WQ_WORDS_E);
     } else {
        VL_MEMSET_W(owp + VL_WQ_WORDS_E, 0, VL_WORDS_I(obits) - VL_WQ_WORDS_E);
     }
```

After writing that maybe it's better to convert val to a bool in VL_MEMSET that selects a hardcoded all zeros or ones, as that will avoid future misuse of non-constant memset values.

Similar below

Reply:
I'm not a fan of the bool argument while also calling it "memset", so I added VL_MEMSET_ZERO_W and VL_MEMSET_ONES_W. with no val argument. If you want a guaranteed const argument, using a template can enforce that, but let's not add that just now for compilation speed reasons.